android/java/org/chromium/chrome/browser/crypto_wallet/util/ImageLoader.java

```diff
@@ -5,47 +5,363 @@
 import java.util.Arrays;
+import java.util.Iterator;
 import java.util.List;
+import java.util.Locale;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.xml.XMLConstants;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
 
 public class ImageLoader {
     private static final List<String> ANIMATED_LIST = Arrays.asList(".gif");
+    private static final String UNUSED_CLIENT_NAME = "unused";
+    private static final String BASE64_ENCODING_PATTERN =
+            "^([A-Za-z0-9+/]{4})*([A-Za-z0-9+/]{4}|[A-Za-z0-9+/]{3}=|[A-Za-z0-9+/]{2}==)$";
+
+    /**
+     * Downloads an image from a given URL, including support for GIF and SVG image types.
+     * @param url URL of the image to download.
+     * @param context Android context used by Glide for applying transformations.
+     * @param isCircular When {@code true}, a circular transformation will be applied.
+     * @param imageView ImageView where the downloaded image will be set.
+     * @param callback Callback used to notify if the image has been set correctly. It can be {@code
+     *         null}.
+     */
+    public static void downloadImage(String url, final Context context, final boolean isCircular,
+            final ImageView imageView, final Callback callback) {
+        if (!isValidImgUrl(url)) {
+            if (callback != null) callback.onLoadFailed();
+            return;
+        }
+
+        Profile profile = Utils.getProfile(false);
+        if (isSvg(url)) {
+            final String validUrl;
+            if (URLUtil.isDataUrl(url)) {
+                if (isBase64Encoded(url)) {
+                    String decodedUrl = decodeBase64SvgUrl(url);
+                    validUrl = sanitizeSvg(decodedUrl);
+                } else if (isUtf8(url)) {
+                    validUrl = sanitizeSvg(url);
+                } else {
+                    // Unsupported URL type.
+                    if (callback != null) callback.onLoadFailed();
+                    return;
+                }
+                if (TextUtils.isEmpty(validUrl)) {
+                    // This may happen for invalid, or corrupted URLs.
+                    if (callback != null) callback.onLoadFailed();
+                    return;
+                }
+            } else {
+                validUrl = url;
+            }
+
+            WebContentsFactory webContentsFactory = new WebContentsFactory();
+            WebContents webContents =
+                    webContentsFactory.createWebContentsWithWarmRenderer(profile, false);
+            webContents.downloadImage(new GURL(validUrl), // Url
+                    true, // isFavIcon - Does not send or accept cookies when `true`.
+                    WalletConstants.MAX_BITMAP_SIZE_FOR_DOWNLOAD, // maxBitmapSize
```

Comment:
This now uses 2048 as bitmap size limt, when over this value the image is scaled automatically. Another option could be using 0, meaning no limits, but this may create issues for big images, draining the battery and provoke memory issues.

Reply:
let's use false for `isFavIcon`